src/rust/en.omegascans/src/parser.rs

```diff
@@ -0,0 +1,406 @@
+use aidoku::{
+	error::Result, prelude::format, std::net::HttpMethod, std::net::Request, std::String, std::Vec,
+	Chapter, Filter, FilterType, Listing, Manga, MangaContentRating, MangaPageResult, MangaStatus,
+	MangaViewer, Page, std::current_date, helpers::uri::encode_uri_component
+};
+
+pub fn parse_manga_list(base_url: String, filters: Vec<Filter>, _page: i32) -> Result<MangaPageResult> {
+	
+	let mut search_query = String::new();
+	let mut status = String::new();
+	let mut genres = String::new();
+	
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				if let Ok(filter_value) = filter.value.as_string() {
+					search_query.push_str(&encode_uri_component(filter_value.read()));
+				}
+			}
+			FilterType::Genre => {
+				if let Ok(filter_id) = filter.object.get("id").as_string() {
+					match filter.value.as_int().unwrap_or(-1) {
+						1 => {
+							genres.push_str(filter_id.read().as_str());
+							genres.push(',');
+						}
+						_ => continue,
+					}
+				}
+			}
+			FilterType::Select => match filter.name.as_str() {
+				"Status" => match filter.value.as_int().unwrap_or(-1) {
+					0 => status.push_str("All"),
+					1 => status.push_str("Ongoing"),
+					2 => status.push_str("Hiatus"),
+					3 => status.push_str("Dropped"),
+					4 => status.push_str("Completed"),
+					5 => status.push_str("Cancelled"),
+					_ => continue,
+				},
+				_ => continue,
+			}
+			_ => continue,
+		}
+	}
+
+	if !genres.is_empty() {
+		genres.pop();
+	}
+
+	let url = format!("https://api.omegascans.org/query?query_string={}&series_status={}&order=desc&orderBy=total_views&series_type=Comic&page=1&perPage=1000&tags_ids=[{}]", search_query, status, genres);
+
+	let json = Request::new(url, HttpMethod::Get)
+		.json()
+		.expect("Failed to load JSON")
```

Comment:
don't use `expect` or `unwrap` unless you know for a fact it won't ever be triggered. use `?` to return the error instead.